sdk/communication/Azure.Communication.Common/src/EntraTokenCredential.cs

```diff
@@ -0,0 +1,145 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace Azure.Communication
+{
+    /// <summary>
+    /// Represents a credential that exchanges an Entra token for an Azure Communication Services (ACS) token, enabling access to ACS resources.
+    /// </summary>
+    internal sealed class EntraTokenCredential : ICommunicationTokenCredential
+    {
+        private HttpPipeline _pipeline;
+        private string _resourceEndpoint;
+        private readonly ThreadSafeRefreshableAccessTokenCache _accessTokenCache;
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="EntraTokenCredential"/>.
+        /// </summary>
+        /// <param name="options">The options for how the token will be fetched</param>
+        /// <param name="pipelineTransport">Only for testing.</param>
+        public EntraTokenCredential(EntraCommunicationTokenCredentialOptions options, HttpPipelineTransport pipelineTransport = null)
```

Comment:
Can we make this constructor with the HttpPipelineTransport override internal, and add a public constructor that only takes in the options?

Reply:
EntraTokenCredential is an internal class, so that's why I went with having optional param instead of creating override. So can override it if you still suggest to override it 😁

Reply:
Ah, makes sense, I forgot that the whole class is internal.